src/oop_ext/foundation/odict.py

```diff
@@ -1,8 +1,11 @@
 import collections
+from typing import Any, Iterable
 
 
-class _OrderedDict(collections.OrderedDict):
-    def insert(self, index, key, value, dict_setitem=dict.__setitem__):
+class odict(collections.OrderedDict):
+    def insert(
+        self, index: int, key: Any, value: Any, dict_setitem: Any = dict.__setitem__
```

Comment:
perhaps `key` could be `typing.Hashable`?

Reply:
Done!